wolfcrypt/src/memory.c

```diff
@@ -949,10 +949,17 @@ void* wolfSSL_Malloc(size_t size, void* heap, int type)
                             break;
                         }
                     #ifdef WOLFSSL_DEBUG_STATIC_MEMORY
+                        #ifdef WOLFSSL_ZEPHYR
+                        else {
+                            fprintf(stderr, "Size: %zu, Empty: %d\n", size,
+                                                              mem->sizeList[i]);
+                        }
+                        #else
                         else {
                             fprintf(stderr, "Size: %ld, Empty: %d\n", size,
                                                               mem->sizeList[i]);
                         }
+                        #endif
```

Comment:
Length specifier is C99, not just Zephyr.  This shows we've had a minor bug here for a while.  Recommend to simply cast to an unsigned long instead to correct the standing bug, recognizing we'll print the wrong size for large allocations.
```suggestion
                        else {
                            fprintf(stderr, "Size: %lu, Empty: %d\n", (unsigned long) size,
                                                              mem->sizeList[i]);
                        }
```